fragmenter/chemi.py

```diff
@@ -1,1619 +1,525 @@
-    else:
-        molecule = oechem.OEMol(molecule)
-
-    charged_copy = generate_conformers(molecule, max_confs=max_confs, strict_stereo=strict_stereo, **kwargs)  # Generate up to max_confs conformers
-
-    # fix issue that causes carboxylic acid to fail charging
-    carboxylic_acid_hack(charged_copy)
-
-    if not legacy:
-        # 2017.2.1 OEToolkits new charging function
-        status = oequacpac.OEAssignCharges(charged_copy, oequacpac.OEAM1BCCCharges())
-        if not status: raise(RuntimeError("OEAssignCharges failed."))
-    else:
-        # AM1BCCSym recommended by Chris Bayly to KAB+JDC, Oct. 20 2014.
-        status = oequacpac.OEAssignPartialCharges(charged_copy, oequacpac.OECharges_AM1BCCSym)
-        if not status: raise(RuntimeError("OEAssignPartialCharges returned error code %d" % status))
-
-    #Determine conformations to return
-    if keep_confs is None:
-        #If returning original conformation
-        original = molecule.GetCoords()
-        #Delete conformers over 1
-        for k, conf in enumerate( charged_copy.GetConfs() ):
-            if k > 0:
-                charged_copy.DeleteConf(conf)
-        #Copy coordinates to single conformer
-        charged_copy.SetCoords( original )
-    elif keep_confs > 0:
-        logger().debug("keep_confs was set to %s. Molecule positions will be reset." % keep_confs)
-
-        #Otherwise if a number is provided, return this many confs if available
-        for k, conf in enumerate( charged_copy.GetConfs() ):
-            if k > keep_confs - 1:
-                charged_copy.DeleteConf(conf)
-    elif keep_confs == -1:
-        #If we want all conformations, continue
-        pass
-    else:
-        #Not a valid option to keep_confs
-        raise(ValueError('Not a valid option to keep_confs in get_charges.'))
-
-    return charged_copy
-
-
-def generate_conformers(molecule, max_confs=800, dense=False, strict_stereo=True, ewindow=15.0, rms_threshold=1.0, strict_types=True,
-                        can_order=True, copy=True, timeout=100):
-    """Generate conformations for the supplied molecule
-    Parameters
-    ----------
-    molecule : OEMol
-        Molecule for which to generate conformers
-    max_confs : int, optional, default=800
-        Max number of conformers to generate.  If None, use default OE Value.
-    strict_stereo : bool, optional, default=True
-        If False, permits smiles strings with unspecified stereochemistry.
-    strict_types : bool, optional, default=True
-        If True, requires that Omega have exact MMFF types for atoms in molecule; otherwise, allows the closest atom
-        type of the same element to be used.
     Returns
     -------
-    molcopy : OEMol
-        A multi-conformer molecule with up to max_confs conformers.
-    Notes
-    -----
-    Roughly follows
-    http://docs.eyesopen.com/toolkits/cookbook/python/modeling/am1-bcc.html
+        A molecule which contains ELF10 AM1 Wiberg Bond orders.
```

Comment:
```suggestion
        A new molecule which contains ELF10 AM1 Wiberg Bond orders.
```

It saves users having to look at code to work out if modifications are done in-place or on a copy.

Reply:
I think there is an implicit contract that if a method returns something like the input then that returned value will be something new (i.e. modifications aren't in place), because it doesn't make too much sense to return to the user what they gave to you. 

Happy to accept the change however to make this extra clear.